solidago/src/solidago/preference_learning/generalized_bradley_terry.py

```diff
@@ -46,23 +47,25 @@ def cumulant_generating_function_derivative(self) -> Callable[[npt.NDArray], npt
         -------
         out: float
         """
-        raise NotImplementedError
-    
+
+    @property
     @abstractmethod
-    def cumulant_generating_function_second_derivative(self, score_diff: float) -> float:
-        """ We estimate uncertainty by the flatness of the negative log likelihood,
-        which is directly given by the second derivative of the cumulant generating function.
-        
-        Parameters
-        ----------
-        score_diff: float
-            Score difference
-            
-        Returns
-        -------
-        out: float
+    def log_likelihood_function(self) -> Callable[[npt.NDArray, npt.NDArray], float]:
+        """The loss function definition is used only to compute uncertainties.
         """
-        pass
+
+    @cached_property
+    def loss_increase_to_solve(self):
```

Comment:
minor: Naming `translated_negative_log_likelihood` (what it is, not what it is meant to be used for) + it's a log likelihood, not a loss